backend/lcfs/web/api/compliance_report/services.py

```diff
@@ -207,39 +275,72 @@ async def get_compliance_reports_paginated(
             reports=reports,
         )
 
-    def _mask_report_status(self, reports: List) -> List:
-        for report in reports:
-            if isinstance(report, ComplianceReportViewSchema):
-                statuses = {
-                    ComplianceReportStatusEnum.Recommended_by_analyst.underscore_value(),
-                    ComplianceReportStatusEnum.Recommended_by_manager.underscore_value(),
-                }
-                if report.report_status in statuses:
-                    report.report_status, report.report_status_id = (
-                        ComplianceReportStatusEnum.Submitted.value,
-                        None,
-                    )
-            elif isinstance(report, ComplianceReportBaseSchema):
-                statuses = {
-                    ComplianceReportStatusEnum.Recommended_by_analyst.value,
-                    ComplianceReportStatusEnum.Recommended_by_manager.value,
-                }
-                if report.current_status.status in statuses:
-                    (
-                        report.current_status.status,
-                        report.current_status.compliance_report_status_id,
-                    ) = (
-                        ComplianceReportStatusEnum.Submitted.value,
-                        None,
-                    )
+    def _mask_report_status(self, reports: List, user: UserProfile) -> List:
```

Comment:
I like the role refactoring you've done in this pr. We just need to make sure we cover our bases with tests for this updated logic.